db/write_batch.cc

```diff
@@ -2612,11 +2609,116 @@ Status WriteBatchInternal::InsertInto(
   return s;
 }
 
+// This class computes checksums for a WriteBatch.
+// This is WIP -- gonzo does not really fully know (yet) what he is doing.
+class ChecksumUpdater : public WriteBatch::Handler {
+ public:
+  explicit ChecksumUpdater(WriteBatch::ProtectionInfo* prot_info)
+      : prot_info_(prot_info) {}
+
+  ~ChecksumUpdater() override {}
+
+  Status PutCF(uint32_t cf, const Slice& key, const Slice& val) override {
+    return update_checksum(cf, key, val);
+  }
+
+  Status DeleteCF(uint32_t /* cf */, const Slice& /* key */) override {
+    // TODO gonzo: does it make sense to checksum the key (no value)?
+    return do_nothing();
+  }
+
+  Status SingleDeleteCF(uint32_t /* cf */, const Slice& /* key */) override {
+    // TODO gonzo: does it make sense to checksum the key (no value)?
+    return do_nothing();
+  }
+
+  Status DeleteRangeCF(uint32_t /* cf */, const Slice& /* begin_key */,
+                       const Slice& /* end_key */) override {
+    // TODO gonzo: does it make sense to checksum the keys (no values)?
```

Comment:
For this, the key can be `begin_key` and the value can be `end_key`: https://github.com/facebook/rocksdb/blob/a5e51305566c3b1d809b1e569850354885d646d5/db/write_batch.cc#L1216